skyportal/plot.py

```diff
@@ -511,6 +577,60 @@ def photometry_plot(obj_id, user, width=600, height=300, device="browser"):
             )
         )
 
+    # Mark when spectra were taken
+    s_mjd = []
+    s_y = []
+    s_date = []
+    s_tel = []
+    s_inst = []
+    s_text = []
+    text_y = (ymax - ymin) * 0.05 + ymin
+    for s in spectra:
+        date = '%d-%d-%dT%d:%d:%d' % (
+            s.observed_at.year,
+            s.observed_at.month,
+            s.observed_at.day,
+            s.observed_at.hour,
+            s.observed_at.minute,
+            s.observed_at.second,
+        )
+        s_mjd.append(Time(date).mjd)
+        s_y.append(text_y)
+        s_date.append(date)
+        s_tel.append(s.instrument.telescope.name)
+        s_inst.append(s.instrument.name)
+        s_text.append("S")
+    if len(s_mjd) > 0:
+        spec_r_mag = plot.text(
+            x='s_mjd',
+            y='s_y',
+            text='s_text',
+            text_alpha=0.3,
+            text_align='center',
+            source=ColumnDataSource(
+                data=dict(
+                    s_mjd=s_mjd,
+                    s_y=s_y,
+                    s_date=s_date,
+                    s_tel=s_tel,
+                    s_inst=s_inst,
+                    s_text=s_text,
+                )
+            ),
+        )
+        plot.add_tools(
```

Comment:
Condense to a single block.

Reply:
I don't understand exactly what you mean.  These are on two separate plots: Mag and Flux.  Do you mean make a function that I can call from both?  That would be one way to combine these.  I would suggest that the more explicit your recommendations are, the better.  Don't assume I am as experienced as you are.

Reply:
So I did make it a function and it works well.  BTW, we could probably do the same thing with the marking of the first and last detections.